### PR TITLE
GOB: Adibou 2 scripts support in degob

### PR DESCRIPTION
**Pull request by sdelamarre**
This PR adds support of Adibou 2 scripts in degob (the decompiler for GOB script files), by porting new opcodes/missing cases in existing opcodes from the main scummvm repo. Also ports the handling of IDE files (named function offsets), with a new --lib mode to use all named function offsets as entry points for degob.

### CODE REVIEW

File: engines/gob/degob.cpp [hunk 1]
```diff
@@ -199,13 +230,16 @@ Script *initScript(byte *totData, uint32 totSize, ExtTable *extTable, int versio
 			break;
 		case 9:
 			return new Script_v6(totData, totSize, extTable);
-			break;
+		break;
```
Comment (BLooperZ) [inline engines/gob/degob.cpp:233]: Indentation is inconsistent here

File: engines/gob/degob_script_v1.cpp [hunk 2]
```diff
@@ -926,13 +926,72 @@ void Script_v1::o1_palLoad(FuncParams &params) {
 		break;
 
 	case 53:
+		skip(2);
+		return;
```
Comment (BLooperZ) [inline engines/gob/degob_script_v1.cpp:930]: Just making sure the return here is intentional, previous version had break here (and this is shared)

**Review by BLooperZ — COMMENTED** [review]
Thank you so much for the amazing work.
Please see some comments on code

File: engines/gob/degob_script_v7.cpp [hunk 3]
```diff
@@ -0,0 +1,784 @@
...
+	} while (peekUint8() != 200);
+
+	endFunc();
+
+	skip(1);
+}
+
+void Script_v7::o7_oemToANSI(FuncParams &params)
+{
+	skip(2);
```
Comment (BLooperZ) [inline engines/gob/degob_script_v7.cpp:773]: Shouldn't this command be printed out?

File: engines/gob/degob_script_v1.cpp [hunk 4]
```diff
@@ -926,13 +926,72 @@ void Script_v1::o1_palLoad(FuncParams &params) {
...
+	}
+
+	switch (cmd) {
+	case 50:
+		for (int i = 0; i < 16; i++)
+			readUint8();
+		break;
+
+		case 52:
```
Comment (BLooperZ) [inline engines/gob/degob_script_v1.cpp:979]: Indentation

File: engines/gob/degob_script_v1.cpp [hunk 5]
```diff
@@ -926,13 +926,72 @@ void Script_v1::o1_palLoad(FuncParams &params) {
...
+			for (int i = 0; i < 16; i++) {
+				readUint8();
+				readUint8();
+				readUint8();
+			}
```
Comment (BLooperZ) [inline engines/gob/degob_script_v1.cpp:984]: 48 bytes were already read above, this seems to break INTRO.TOT in Gobliiins 3

**Comment by sdelamarre** [comment]
Thanks a lot for the valuable input @BLooperZ. The regression on o1_palLoad should be fixed now, as well as the other small issues.

File: engines/gob/degob_script_v1.cpp (hunk 5 above)
Comment (sdelamarre) [reply engines/gob/degob_script_v1.cpp]: Indeed ! The o1_palLoad was not only broken, but most of the added code was not needed at all (I tried to mimic some logic of the o1_palLoad from the main scummvm repo without seeing for degob it gets simplified to just a matter of skipping the right amount of bytes depending on the cmd value). I could reproduced the regression it also on the GOB2 demo, it is fixed now.

File: engines/gob/degob_script_v7.cpp [hunk 6]
```diff
@@ -0,0 +1,786 @@
+/* ScummVM Tools
+ *
+ * ScummVM Tools is the legal property of its developers, whose
+ * names are too numerous to list here. Please refer to the
+ * COPYRIGHT file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
```
Comment (sev-) [inline engines/gob/degob_script_v7.cpp:9]: This is incorrect license. Please change to GPLv3+

Comment (sdelamarre) [reply engines/gob/degob_script_v7.cpp]: Now fixed.

**Comment by sev-** [comment]
Thank you!

**Pull request merged**